ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PGSchema.scala

```diff
@@ -0,0 +1,142 @@
+    "create_key_value_compression" -> PGSmallintOptional(_.create_key_value_compression),
+  )
+
+  val exerciseFields: Vector[(String, PGField[DBDTOV1.EventExercise, _, _])] =
+    Vector[(String, PGField[DBDTOV1.EventExercise, _, _])](
+      "event_id" -> PGString(_.event_id.orNull),
+      "event_offset" -> PGString(_.event_offset.orNull),
+      "contract_id" -> PGString(_.contract_id),
+      "transaction_id" -> PGString(_.transaction_id.orNull),
+      "ledger_effective_time" -> PGTimestamp(_.ledger_effective_time.orNull),
+      "node_index" -> PGIntOptional(_.node_index),
+      "command_id" -> PGString(_.command_id.orNull),
+      "workflow_id" -> PGString(_.workflow_id.orNull),
+      "application_id" -> PGString(_.application_id.orNull),
+      "submitters" -> PGStringArray(_.submitters.orNull),
+      "create_key_value" -> PGBytea(_.create_key_value.orNull),
+      "exercise_choice" -> PGString(_.exercise_choice.orNull),
+      "exercise_argument" -> PGBytea(_.exercise_argument.orNull),
+      "exercise_result" -> PGBytea(_.exercise_result.orNull),
+      "exercise_actors" -> PGStringArray(_.exercise_actors.orNull),
+      "exercise_child_event_ids" -> PGStringArray(_.exercise_child_event_ids.orNull),
+      "template_id" -> PGString(_.template_id.orNull),
+      "flat_event_witnesses" -> PGStringArray(_.flat_event_witnesses),
+      "tree_event_witnesses" -> PGStringArray(_.tree_event_witnesses),
+      "event_sequential_id" -> PGBigint(_.event_sequential_id),
+      "create_key_value_compression" -> PGSmallintOptional(_.create_key_value_compression),
+      "exercise_argument_compression" -> PGSmallintOptional(_.exercise_argument_compression),
+      "exercise_result_compression" -> PGSmallintOptional(_.exercise_result_compression),
+    )
+
+  val eventsConsumingExercise: PGTable[DBDTOV1.EventExercise] =
+    PGTable(tableName = "participant_events_consuming_exercise", exerciseFields)
+
+  val eventsNonConsumingExercise: PGTable[DBDTOV1.EventExercise] =
+    PGTable(tableName = "participant_events_non_consuming_exercise", exerciseFields)
+
+  val configurationEntries: PGTable[DBDTOV1.ConfigurationEntry] = PGTable("configuration_entries")(
+    "ledger_offset" -> PGString(_.ledger_offset),
+    "recorded_at" -> PGTimestamp(_.recorded_at),
+    "submission_id" -> PGString(_.submission_id),
+    "typ" -> PGString(_.typ),
+    "configuration" -> PGBytea(_.configuration),
+    "rejection_reason" -> PGString(_.rejection_reason.orNull),
+  )
+
+  val packageEntries: PGTable[DBDTOV1.PackageEntry] = PGTable("package_entries")(
+    "ledger_offset" -> PGString(_.ledger_offset),
+    "recorded_at" -> PGTimestamp(_.recorded_at),
+    "submission_id" -> PGString(_.submission_id.orNull),
+    "typ" -> PGString(_.typ),
+    "rejection_reason" -> PGString(_.rejection_reason.orNull),
+  )
+
+  val packages: PGTable[DBDTOV1.Package] = PGTable(
+    tableName = "packages",
+    insertSuffix = "on conflict (package_id) do nothing",
```

Comment:
This insert conflict action clause is not a property of a table but rather of a statement. I see this is also the only place where it is non-empty. I'd remove it from `PGTable` if it doesn't complicate code in other places much

Reply:
I agree in general, but here in this DSL PGTable's responsibility is to:
* provide method to create the insert SQL for preparedstatements (val insertStatement)
* provide the method to assemble batch arrays (def prepareData)
* provide method to set up data preparedStatement data from built batch arrays (def setupData)

We could of course walk some more lengths to have these responsibilities separated somehow, currently it seemed to me the right level of abstraction, but please hit me with ideas if you can think something better!